internal/mtls/caprovider_secret.go

```diff
@@ -118,3 +127,55 @@ func (config *CASecretProvider) CreateRegistrationCertificate(name string) (map[
 	}
 	return res, nil
 }
+
+// SignCSR sign a new CertificateRequest and returns the PEM certificate.
+// This function is going to be used a lot, so using config.latestCA ensure
+// that APIServer is not overloaded with that.
+// Because the CM is always managed by this, should be safe to use that one.
+func (config *CASecretProvider) SignCSR(CSRPem string, commonName string, expiration time.Time) ([]byte, error) {
+	if config.latestCA == nil {
+		return nil, fmt.Errorf("Cannot get CA certificate")
+	}
+	// next blocks to be avoided just because we only sign one CSR. If more than
+	// one maybe it's an attack.
+	decodecCert, _ := pem.Decode([]byte(CSRPem))
+	if decodecCert == nil {
+		return nil, fmt.Errorf("Cannot decode CSR certificate")
+	}
+
+	CSR, err := x509.ParseCertificateRequest(decodecCert.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse CSR: %v", err)
+	}
+
+	clientCert := &x509.Certificate{
+		Signature:          CSR.Signature,
+		SignatureAlgorithm: CSR.SignatureAlgorithm,
+		PublicKeyAlgorithm: CSR.PublicKeyAlgorithm,
+		PublicKey:          CSR.PublicKey,
+		SerialNumber:       big.NewInt(time.Now().Unix()),
+		Subject:            CSR.Subject,
+		NotBefore:          time.Now(),
```

Comment:
In the past this was problematic due to time drift on different machines. We do not require ntpd nor chronyd. In the past we decided to set it to `time.Now() - 24h` to mitigate `certificate not valid yet`.